test/functional/mining_testnet.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test mining on testnet
+
+Test mining related RPCs that involve difficulty adjustment, which
+regtest doesn't have.
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+from test_framework.blocktools import (
+    DIFF_1_N_BITS,
+    DIFF_1_TARGET,
+    DIFF_4_N_BITS,
+    DIFF_4_TARGET,
+    nbits_str,
+    target_str
+)
+
+import os
+
+class MiningTestnetTest(BitcoinTestFramework):
+
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+        self.chain = "testnet4"
```

Comment:
In the future, when testnet4 replaces testnet3 as `-testnet` (no trailing number), we'll need to drop the `4`.

Reply:
Maybe, depends on how soon testnet5 shows up :-)